code/modules/surgery/surgery.dm

```diff
@@ -99,18 +99,23 @@
 		return
 	surgery.step_in_progress = 1
 
+	var/speed_mod = 1
+
 	if(begin_step(user, target, target_zone, tool, surgery) == -1)
 		surgery.step_in_progress = 0
 		return
 
-	var/advance = 0
-	var/prob_chance = 100
+	if(tool)
+		speed_mod = tool.toolspeed
+
+	if(do_after(user, time * speed_mod, target = target))
+		var/advance = 0
+		var/prob_chance = 100
+
+		if(implement_type)	//this means it isn't a require nd or any item step.
+			prob_chance = allowed_tools[implement_type]
```

Comment:
no `min` necessary now that all tools top out at 100 and the fact we may want to genuinely have tools that have greater than 100 chance of success.